crates/astria-core/src/crypto.rs

```diff
@@ -0,0 +1,83 @@
+use std::fmt::{
+    self,
+    Debug,
+    Formatter,
+};
+
+use ed25519_consensus::{
+    Signature,
+    SigningKey as Ed25519SigningKey,
+    VerificationKey,
+    // VerificationKeyBytes,
+};
+use rand::{
+    CryptoRng,
+    RngCore,
+};
+use zeroize::{
+    Zeroize,
+    ZeroizeOnDrop,
+};
+
+#[derive(Clone, Zeroize, ZeroizeOnDrop)]
+pub struct SigningKey(Ed25519SigningKey);
```

Comment:
Small doc comment on top, and also a (non-doc) comment to explain its purpose (I thought it prudent to not mention it in the doc comment for the same reason that you decided to not expose a third-party crate).
```suggestion
/// An Ed25519 signing key.
// *Implementation note*: this is currently a refinement type around
// ed25519_consensus::SigningKey overriding its Debug implementation
// to not accidentally leak it.
pub struct SigningKey(Ed25519SigningKey);
```

Reply:
Done in [6b05e11](https://github.com/astriaorg/astria/pull/1074/commits/6b05e114d3a3860868252108f2fc7be22636e433).